go.mod

```diff
@@ -1,114 +1,113 @@
 module k8c.io/operating-system-manager
 
-go 1.20
+go 1.21
 
 require (
 	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/Masterminds/sprig/v3 v3.2.3
-	github.com/flatcar-linux/container-linux-config-transpiler v0.9.3
-	github.com/go-logr/logr v1.2.4
+	github.com/flatcar/container-linux-config-transpiler v0.9.4
+	github.com/go-logr/zapr v1.3.0
 	github.com/go-test/deep v1.1.0
-	github.com/kubermatic/machine-controller v1.57.0
-	github.com/onsi/ginkgo v1.16.5
+	github.com/kubermatic/machine-controller v1.58.0
+	github.com/onsi/ginkgo/v2 v2.13.1
 	github.com/pmezard/go-difflib v1.0.0
-	go.uber.org/zap v1.24.0
+	github.com/spf13/pflag v1.0.6-0.20210604193023-d5e0c0615ace
+	go.uber.org/zap v1.26.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8c.io/reconciler v0.3.1
-	k8s.io/api v0.26.4
-	k8s.io/apimachinery v0.26.4
+	k8c.io/reconciler v0.4.0
+	k8s.io/api v0.28.4
+	k8s.io/apimachinery v0.28.4
 	k8s.io/client-go v12.0.0+incompatible
```

Comment:
Please switch this to v0.28.4 :)

Reply:
Just removing the replace directive doesn't work:

```
▶ go mod tidy  
go: finding module for package k8s.io/client-go/applyconfigurations/meta/v1
go: finding module for package k8s.io/api/settings/v1alpha1
go: finding module for package k8s.io/api/auditregistration/v1alpha1
go: finding module for package k8s.io/api/batch/v2alpha1
go: finding module for package k8s.io/client-go/metadata
go: finding module for package k8s.io/apimachinery/pkg/util/clock
go: k8c.io/operating-system-manager/cmd/osm-controller imports
        k8s.io/client-go/kubernetes/scheme imports
        k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.28.4), but does not contain package k8s.io/api/auditregistration/v1alpha1
go: k8c.io/operating-system-manager/cmd/osm-controller imports
        k8s.io/client-go/kubernetes/scheme imports
        k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.28.4), but does not contain package k8s.io/api/batch/v2alpha1
go: k8c.io/operating-system-manager/cmd/osm-controller imports
        k8s.io/client-go/kubernetes/scheme imports
        k8s.io/api/settings/v1alpha1: module k8s.io/api@latest found (v0.28.4), but does not contain package k8s.io/api/settings/v1alpha1
go: k8c.io/operating-system-manager/cmd/osm-controller imports
        sigs.k8s.io/controller-runtime/pkg/client imports
        k8s.io/client-go/metadata: package k8s.io/client-go/metadata provided by k8s.io/client-go at latest version v0.28.4 but not at required version v12.0.0+incompatible
go: k8c.io/operating-system-manager/pkg/controllers/osc imports
        k8s.io/client-go/tools/record imports
        k8s.io/apimachinery/pkg/util/clock: module k8s.io/apimachinery@latest found (v0.28.4), but does not contain package k8s.io/apimachinery/pkg/util/clock
go: k8c.io/operating-system-manager/cmd/osm-controller imports
        sigs.k8s.io/controller-runtime/pkg/cache tested by
        sigs.k8s.io/controller-runtime/pkg/cache.test imports
        sigs.k8s.io/controller-runtime/pkg/envtest imports
        k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset imports
        k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1 imports
        k8s.io/apiextensions-apiserver/pkg/client/applyconfiguration/apiextensions/v1 imports
        k8s.io/client-go/applyconfigurations/meta/v1: package k8s.io/client-go/applyconfigurations/meta/v1 provided by k8s.io/client-go at latest version v0.28.4 but not at required version v12.0.0+incompatible
```

It's something benign so I don't really want to spend more time on it in this PR :D Maybe later sometime.

Reply:
> go: k8s.io/client-go@v12.0.0+incompatible: invalid version: module contains a go.mod file, so module path must match major version ("k8s.io/client-go/v12")

Reply:
Funny enough, the PR in KKP that you approved had the solution: reconciler 0.5.0. :) If you update the dependency, you can fix the client-go dependency and remove the replace directive. After a `go mod tidy` my diff would be

```diff
diff --git a/go.mod b/go.mod
index 94e0830..00ed5d4 100644
--- a/go.mod
+++ b/go.mod
@@ -14,13 +14,13 @@ require (
        github.com/spf13/pflag v1.0.6-0.20210604193023-d5e0c0615ace
        go.uber.org/zap v1.26.0
        gopkg.in/yaml.v3 v3.0.1
-       k8c.io/reconciler v0.4.0
+       k8c.io/reconciler v0.5.0
        k8s.io/api v0.28.4
        k8s.io/apimachinery v0.28.4
-       k8s.io/client-go v12.0.0+incompatible
+       k8s.io/client-go v0.28.4
        k8s.io/code-generator v0.28.4
        k8s.io/klog/v2 v2.110.1
-       k8s.io/utils v0.0.0-20231121161247-cf03d44ff3cf
+       k8s.io/utils v0.0.0-20231127182322-b307cd553661
        sigs.k8s.io/controller-runtime v0.16.3
        sigs.k8s.io/controller-tools v0.13.0
        sigs.k8s.io/yaml v1.4.0
@@ -108,6 +108,4 @@ require (
        sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )

-replace k8s.io/client-go => k8s.io/client-go v0.28.4
-
 replace github.com/ajeddeloh/go-json => github.com/coreos/go-json v0.0.0-20230131223807-18775e0fb4fb
```

Reply:
That worked, updated the PR. Can you take another look? ;)